### PR TITLE
feat: add support for Kubernetes 1.21.0-beta.1

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:

See https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.21.md#changelog-since-v1210-beta0

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

**Requirements**:

- [x] Kubernetes artifacts built and pushed by Azure Pipelines
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Review by jackfrancis — DISMISSED** [review]
/lgtm

**Comment by jackfrancis** [comment]
@marosset @jsturtevant 1.21.0-beta.1 + containerd looks to have a regression compared to beta.0:

```
Warning  Failed     30m   kubelet            Error: failed to create containerd task: hcsshim::CreateComputeSystem validate-bing: The directory name is invalid.
(extra info: {"Owner":"containerd-shim-runhcs-v1.exe","SchemaVersion":{"Major":2,"Minor":1},"Container":{"GuestOs":{"HostName":"validate-dns-windows-tdhkg"},"Storage":{"Layers":[{"Id":"fd99f5a1-d5be-57d1-ac12-5f04d4e788e3","Path":"C:\\ProgramData\\containerd\\root\\io.containerd.snapshotter.v1.windows\\snapshots\\2"},{"Id":"65ed1907-894d-50ff-bdcb-cc63fb515966","Path":"C:\\ProgramData\\containerd\\root\\io.containerd.snapshotter.v1.windows\\snapshots\\1"}],"Path":"\\\\?\\Volume{5959079b-7046-47ee-aae5-5428de968dcf}\\"},"MappedDirectories":[{"HostPath":"c:\\var\\lib\\kubelet\\pods\\9c9dff15-eff4-4c54-9b84-4ab498a541bc\\containers\\validate-bing\\527f3363","ContainerPath":"c:\\dev\\termination-log"},{"HostPath":"c:\\var\\lib\\kubelet\\pods\\9c9dff15-eff4-4c54-9b84-4ab498a541bc\\volumes\\kubernetes.io~projected\\kube-api-access-wbwdx","ContainerPath":"c:\\var\\run\\secrets\\kubernetes.io\\serviceaccount","ReadOnly":true},{"HostPath":"c:\\var\\lib\\kubelet\\pods\\9c9dff15-eff4-4c54-9b84-4ab498a541bc\\etc-hosts","...
  Normal   Pulled   30m  kubelet  Container image "mcr.microsoft.com/windows/servercore:ltsc2019" already present on machine
  Normal   Created  30m  kubelet  Created container validate-google
  Warning  Failed   30m  kubelet  Error: failed to create containerd task: hcsshim::CreateComputeSystem validate-google: The directory name is invalid.
(extra info: {"Owner":"containerd-shim-runhcs-v1.exe","SchemaVersion":{"Major":2,"Minor":1},"Container":{"GuestOs":{"HostName":"validate-dns-windows-tdhkg"},"Storage":{"Layers":[{"Id":"fd99f5a1-d5be-57d1-ac12-5f04d4e788e3","Path":"C:\\ProgramData\\containerd\\root\\io.containerd.snapshotter.v1.windows\\snapshots\\2"},{"Id":"65ed1907-894d-50ff-bdcb-cc63fb515966","Path":"C:\\ProgramData\\containerd\\root\\io.containerd.snapshotter.v1.windows\\snapshots\\1"}],"Path":"\\\\?\\Volume{5959079b-7046-47ee-aae5-5428de968dcf}\\"},"MappedDirectories":[{"HostPath":"c:\\var\\lib\\kubelet\\pods\\9c9dff15-eff4-4c54-9b84-4ab498a541bc\\containers\\validate-google\\a59acf60","ContainerPath":"c:\\dev\\termination-log"},{"HostPath":"c:\\var\\lib\\kubelet\\pods\\9c9dff15-eff4-4c54-9b84-4ab498a541bc\\volumes\\kubernetes.io~projected\\kube-api-access-wbwdx","ContainerPath":"c:\\var\\run\\secrets\\kubernetes.io\\serviceaccount","ReadOnly":true},{"HostPath":"c:\\var\\lib\\kubelet\\pods\\9c9dff15-eff4-4c54-9b84-4ab498a541bc\\etc-hosts"...
```

Any thoughts?

**Comment by marosset** [comment]
https://github.com/Azure/aks-engine/blob/b24cda7e738e30b45323fbd3a330bd66a61ecea1/test/e2e/test_cluster_configs/containerd.json#L54 needs to be updated to `2019-datacenter-core-smalldisk-containerd-2103`

Do you want to include that change in this PR or should I open another one?

**Comment by jackfrancis** [comment]
@marosset I'll make the change here, yay for the clear resolution!

**Comment by marosset** [comment]
We enabled some functionality in k/k for contianer that that requires the Windows nodes have at least the Oct 2020 security patches. The image being used by the e2e tests here has May 2020 security patches :(

**Review by marosset — DISMISSED** [review]
/lgtm

File: test/e2e/engine/template.go [hunk 1]
```diff
@@ -261,7 +261,7 @@ func Build(cfg *config.Config, masterSubnetID string, agentSubnetIDs []string, i
 				prop.WindowsProfile.ImageVersion == "" {
 				prop.WindowsProfile.WindowsPublisher = "microsoft-aks"
 				prop.WindowsProfile.WindowsOffer = "aks-windows"
-				prop.WindowsProfile.WindowsSku = "2019-datacenter-core-smalldisk-containerd-2005"
+				prop.WindowsProfile.WindowsSku = "2019-datacenter-core-smalldisk-containerd-2103"
```
Comment (marosset) [inline test/e2e/engine/template.go:264]: Oops sorry this needs to be 2019-datacenter-core-ctrd-2103

File: test/e2e/test_cluster_configs/containerd.json [hunk 2]
```diff
@@ -51,7 +51,7 @@
                 "sshEnabled": true,
                 "windowsPublisher": "microsoft-aks",
                 "windowsOffer": "aks-windows",
-                "windowsSku": "2019-datacenter-core-smalldisk-containerd-2005",
+                "windowsSku": "2019-datacenter-core-smalldisk-containerd-2103",
```
Comment (marosset) [inline test/e2e/test_cluster_configs/containerd.json:54]: Same as above (2019-datacenter-core-ctrd-2103)

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Pull request merged**